### PR TITLE
Use StorageManager to persist and load the bridge database

### LINKED ISSUE #1905: [Bridge] Connect database to StorageManager actor
Connect the StorageManager to a database engine such as RocksDB. :)

### PR DESCRIPTION
**Pull request by lrubiorod**
Close #1905

### CODE REVIEW

File: bridges/centralized-ethereum/src/actors/dr_database.rs [hunk 1]
```diff
@@ -35,14 +42,38 @@ pub enum DrState {
     Finished,
 }
 
+impl Default for DrState {
```
Comment (tmpolaczyk) [inline bridges/centralized-ethereum/src/actors/dr_database.rs:45]: Why is this needed?

File: bridges/centralized-ethereum/src/actors/dr_database.rs [hunk 2]
```diff
@@ -35,14 +42,38 @@ pub enum DrState {
...
+                            act.max_dr_id = dr_database_from_storage.max_dr_id;
+                        }
+                    }
+                    Err(e) => {
+                        panic!("Error while getting bridge database from storage: {}", e);
+                    }
+                },
+            );
+
+        ctx.spawn(fut);
```
Comment (tmpolaczyk) [inline bridges/centralized-ethereum/src/actors/dr_database.rs:76]: To avoid handling other messages before the database is initialized:

```suggestion
        ctx.wait(fut);
```

File: bridges/centralized-ethereum/src/actors/dr_database.rs [hunk 3]
```diff
@@ -70,11 +101,22 @@ impl Message for GetLastDrId {
...
 
-    fn handle(&mut self, msg: SetDrInfoBridge, _ctx: &mut Self::Context) -> Self::Result {
+    fn handle(&mut self, msg: SetDrInfoBridge, ctx: &mut Self::Context) -> Self::Result {
         let SetDrInfoBridge(dr_id, dr_info) = msg;
         self.dr.insert(dr_id, dr_info);
 
         self.max_dr_id = cmp::max(self.max_dr_id, dr_id);
+
+        // Persist Data Request Database
+        let bridge_db = self.clone();
```
Comment (tmpolaczyk) [inline bridges/centralized-ethereum/src/actors/dr_database.rs:111]: Maybe this clone can be avoided by doing something like this: (not tested)

```rust
let f = storage_mngr::put(&BRIDGE_DB_KEY, &self);
let fut = async move {
    let res = f.await;
}
```

File: bridges/centralized-ethereum/src/actors/dr_database.rs (hunk 1 above)
Comment (lrubiorod) [reply bridges/centralized-ethereum/src/actors/dr_database.rs:57]: image

File: bridges/centralized-ethereum/src/actors/dr_database.rs (hunk 3 above)
Comment (lrubiorod) [reply bridges/centralized-ethereum/src/actors/dr_database.rs:132]: image

File: bridges/centralized-ethereum/src/actors/dr_database.rs (hunk 1 above)
Comment (tmpolaczyk) [reply bridges/centralized-ethereum/src/actors/dr_database.rs:57]: Thanks compiler :bow:

File: bridges/centralized-ethereum/src/actors/dr_database.rs (hunk 3 above)
Comment (tmpolaczyk) [reply bridges/centralized-ethereum/src/actors/dr_database.rs:132]: Interesting, it works if you remove the `&` from `&self`:

    let f = storage_mngr::put(&BRIDGE_DB_KEY, self);

**Review by tmpolaczyk — APPROVED** [review]
